nomad/structs/structs.go

```diff
@@ -2787,18 +2787,20 @@ type NodeNetworkAddress struct {
 }
 
 type AllocatedPortMapping struct {
-	Label  string
-	Value  int
-	To     int
-	HostIP string
+	Label           string
```

Comment:
This will probably fix the plan normalization test, but being even close to the threshold on that test right now suggests we're missing this on a bunch of other objects too.

```suggestion
	// msgpack omit empty fields during serialization
	_struct bool `codec:",omitempty"` // nolint: structcheck
	Label           string
```

(Similar for the `Port` type definition below)

Reply:
I swear I tried this and it didn't help much, but now that I try again, it makes a significant impact (on the TestPlanNormalize test).

It'll mean that lots of inline `Port{"label", 0, 0, ""}`s in tests, which I already added `false` to, will need poking again.  I might just make all of those multi-line with explicit fields while I'm at it.

Reply:
I split this out to #23980 so it can be backported, and should result in this PR being a lot shorter without all the totally tangential test changes.